products/llbuildSwift/Internals.swift

```diff
@@ -46,39 +46,39 @@ internal func stringFromData(_ data: llb_data_t) -> String {
 }
 
 extension Array where Element == String {
-    internal func withCArrayOfStrings<T>(_ body: @escaping (UnsafePointer<UnsafePointer<CChar>>) -> T) -> T {
-        func appendPointer(_ index: Array.Index, to target: inout Array<UnsafePointer<CChar>>) -> T {
-            if index == self.endIndex {
-                return body(&target)
-            } else {
-                return self[index].withCString { cStringPtr in
-                    target.append(cStringPtr)
-                    return appendPointer(self.index(after: index), to: &target)
-                }
+    private func withTemporaryBuffer<T>(_ body: @escaping (UnsafePointer<UnsafePointer<CChar>>) -> T) -> T {
+        // The buffer is in the form "a1\0a2\0a3\0…"
+        let totalLength = reduce(0) { $0 + $1.utf8.count + 1 }
+        let buffer = UnsafeMutableRawPointer.allocate(byteCount: totalLength, alignment: 1)
```

Comment:
Shouldn't alignment be something more principled than 1?

Reply:
Changing this to be `UnsafeMutablePointer<CChar>` instead, which defines the alignment based on the generic type.